lib/engines/hcktest/hcktest.rb

```diff
@@ -225,19 +226,64 @@ def prepare_tests
         return
       end
 
-      @tests.list_tests(log: true)
+      @test_list = @tests.list_tests(log: true)
     end
 
     def auto_run
       ResourceScope.open do |scope|
-        run_and_configure_setup scope
-        prepare_tests
+        run_studio scope
+        sleep 5 until @studio.up?
+
+        run_tests_without_config
+        run_tests_with_config
+
+        @tests.create_project_package
+      end
+    end
 
-        @tests.run
+    def run_tests_without_config
+      ResourceScope.open do |scope|
+        run_clients_and_configure_setup scope
+        prepare_tests
 
+        @logger.info('Environment ready, running normal tests')
+        @tests.run(group_tests[:normal])
         pause_run if @project.options.test.manual
+      end
+    end
 
-        @tests.create_project_package
+    def run_tests_with_config
+      group_tests.each do |group, tests|
+        next if tests.empty? || group == :normal
+
+        ResourceScope.open do |scope|
+          run_clients_and_configure_setup(scope, client_run_options(group))
```

Comment:
I guess you can just do:

```suggestion
          run_clients_and_configure_setup(scope, group => true)
```